rustler_codegen/src/encode_decode_templates.rs

```diff
@@ -69,7 +133,56 @@ pub(crate) fn decoder(ctx: &Context, inner: TokenStream) -> TokenStream {
 
 pub(crate) fn encoder(ctx: &Context, inner: TokenStream) -> TokenStream {
     let ident = ctx.ident;
-    let generics = ctx.generics;
+    let mut generics = ctx.generics.clone();
+    let type_parameters: Vec<_> = generics
+        .params
+        .iter()
+        .filter_map(|g| match g {
+            syn::GenericParam::Type(t) => Some(t.clone()),
+            _ => None,
+        })
+        .collect();
+
+    if !type_parameters.is_empty() {
```

Comment:
I think we can remove this `if` and iterate over an empty `type_parameters` instead. With that, we reduce indentation, which makes this more readable.

Reply:
True, I copied the pattern from the lifetimes clause before.

Reply:
I would remove the corresponding `if` from the lifetimes as well, because if I reduce the indentation I need to unconditionally create a `where` clause.

Reply:
The only other way would be calling `generics.make_where_clause()` inside the loop. I believe this would work because it's idempotent, but I don't know what's the best way.

Reply:
Removing the corresponding if makes sense I guess, the parts should look fairly similar in the end.